ledger/accountdb_test.go

```diff
@@ -1078,6 +1078,60 @@ func BenchmarkWriteCatchpointStagingBalances(b *testing.B) {
 	}
 }
 
+func TestLookupKeysByPrefix(t *testing.T) {
+	dbs, fn := dbOpenTest(t, false)
+	setDbLogging(t, dbs)
+	defer cleanupTestDb(dbs, fn, false)
+
+	// return account data, initialize DB tables from accountsInitTest
+	_ = benchmarkInitBalances(t, 1, dbs, protocol.ConsensusCurrentVersion)
+
+	qs, err := accountsInitDbQueries(dbs.Rdb.Handle)
+	require.NoError(t, err)
+	defer qs.close()
+
+	kvPairTestCases := []struct {
+		key   []byte
+		value []byte
+	}{
+		{key: []byte{0xFF, 0x12, 0x34, 0x56, 0x78}, value: []byte("value0")},
```

Comment:
I recommend expanding this test to encompass more typical and also more edge cases. Some case of interest:

* when the prefix is:
  * the empty string `''` 
  * string that looks like `FF FF FF FE FF FF FF`
  * hohum string such as `"abracadabra"`
* insert some well crafted keys and assert that you get exactly what you expect in the result set.